app/fr/maif/izanami/web/SearchController.scala

```diff
@@ -17,81 +18,100 @@ class SearchController(
 ) extends BaseController {
   implicit val ec: ExecutionContext = env.executionContext
 
-  def search(query: String): Action[AnyContent] = tenantRightAction.async {
-    implicit request: UserRequestWithTenantRights[AnyContent] =>
-      {
-        val tenants = request.user.tenantRights.keySet
-        Future
-          .sequence(
-            tenants
-              .map(tenant =>
-                env.datastores.search
-                  .tenantSearch(tenant, request.user.username, query)
-                  .map(l =>
-                    l.map(t => {
-                      (t._1, t._2, t._3, tenant)
-                    })
-                  )
-              )
-          )
-          .map(l => l.flatten.toList.sortBy(_._3)(Ordering.Double.TotalOrdering.reverse).take(10))
-          .flatMap(results => {
-            Future.sequence(results.map { case (rowType, rowJson, _, tenant) =>
-              buildPath(rowType, rowJson, tenant)
-                .map(pathElements => {
-                  val name     = (rowJson \ "name").asOpt[String].getOrElse("")
-                  val jsonPath =
-                    Json.toJson(pathElements.prepended(TenantPathElement(tenant)))(Writes.seq(pathElementWrite))
-                  Json.obj(
-                    "type"   -> rowType,
-                    "name"   -> name,
-                    "path"   -> jsonPath,
-                    "tenant" -> tenant
-                  )
-                })
+    private def checkSearchParams(query: String, filter: List[String]): Future[Either[IzanamiError, Unit]] = {
+    if (query.isEmpty) {
+      return Future.successful(Left(SearchQueryError()))
+    }
+    val validFilters =
+      List("project", "feature", "key", "tag", "script", "global_context", "local_context", "webhook")
+    if (filter.nonEmpty && !filter.forall(validFilters.contains)) {
+      return Future.successful(Left(SearchFilterError(validFilters.mkString(", "))))
+    }
+    Future.successful(Right())
+  }
+
+  def search(query: String, filter: List[String]): Action[AnyContent] = tenantRightAction.async {
+    implicit request: UserRequestWithTenantRights[AnyContent] => {
+      checkSearchParams(query, filter).flatMap {
+        case Left(error) =>
+          Future.successful(BadRequest(Json.obj("error" -> error.message)))
```

Comment:
if `error` is an `IzanamiError` there is an helper to automatically convert it to an http response, just call `error.toHttpResponse`.